NuCypher_Beachhead.ipynb

```diff
@@ -0,0 +1,606 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "m = 1"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "4) Finally, choose a specific file path for as the 'label' of the data we intend to share. \n",
+    "\n",
+    "*Note: the label variable does not refer to the exact file that we intend to share. Rather, it functions more like a tag, where any amount of data can use the same label – a bit like multiple files in a single directory. As we'll see, this means that the data does not necessarily need to exist at this stage, and can be produced later.*"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "label = b\"secret/files/and/stuff\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Before we can begin granting permissions to anyone, we need Alice to connect to the NMKS network and the Ursulas/proxies that will perform the re-encryption service on her behalf. We bootstrap this process by connecting Alice to one known Ursula, who then connects Alice to other Ursulas in their network. Note: there will be an option to start the connection process with Ursulas run by NuCypher, if this is preferable. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ALICE.network_bootstrap([(\"localhost\", 3601)])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### *Bob*: the designated recipient\n",
+    "Bob is the second character in the data sharing narrative. Conceptually, his role is fairly passive – he is simply the chosen recipient. However, as we shall see, there are a number of more complex actions/requirements this character must perform/fulfill in order to securely gain access to the data. Remember, like Alice, Bob represents the device(s) a real-world data recipient – or in our hypothetical application, the doctor – would use to achieve their goals. \n",
+    "\n",
+    "Right now, we'll just instantiate Bob, so we can add him to Alice's sharing policy in the next section.  "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "BOB = Bob()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Creating a sharing policy with *Alice*\n",
+    "It's time to create our first sharing policy, using the .*grant()* function. Sharing policies have three distinguishing features, which are defined when a policy is granted: \n",
+    "\n",
+    " 1) The data controller, *Alice*.\n",
+    " \n",
+    " 2) The designated recipient, *Bob*.\n",
+    " \n",
+    " 3) The file(s) location, *label*.\n",
```

Comment:
In a previous version of NuCypher, labels were called 'uris' because they represented file locations. The new 'label' concept offers a more general approach to catagorize data, beyond mere location-based categories (i.e., paths). I think this notebook is a good place to start talking about data categories represented by labels. For example, in a healthcare use case, a label could be `cardio`, which covers all Alice's data related to cardiology.

Reply:
@cygnusv I agree, this is exactly the right place to illuminate useful capabilities like this one, and I like the example you mentioned. To fully get the message across, we should explain an example label's structure, and how it can be altered – the one we use here (*b"secret/files/and/stuff"*) doesn't tell you a great deal.